pkg/hubble/exporter/builder.go

```diff
@@ -0,0 +1,75 @@
+// SPDX-License-Identifier: Apache-2.0
+// Copyright Authors of Hubble
+
+package exporter
+
+import (
+	"fmt"
+	"maps"
+	"slices"
+)
+
+// FlowLogExporterBuilder knows how to build a FlowLog The Replaces field
+// indicate that an existing builder with a matching Name field should be
+// replaced by this one.
+type FlowLogExporterBuilder struct {
```

Comment:
Any reason this is a struct and not an interface?

Reply:
It could probably be, but I didn't think we really needed it.

Alternative could be:
```go
type FlowLogExporterBuilder interface {
	Name() string
	Replaces() string
	Build() (FlowLogExporter, error)
}
```
But I feel I would still keep one struct that implements this interface instead of having two types for static and dynamic.

Did you have something else in mind? And I can convert it if wanted, it's not much different from what we have.

Reply:
I see that we are introducing a lot of logic in `newHubbleStaticExporter()` and in `newHubbleDynamicExporter()` were we create two different `FlowLogExporterBuilder`. There is actually no logic implemented in `pkg/hubble/exporter` where the struct is defined, it's all implemented in `pkg/hubble/cell` within the two aforementioned functions. So we're building logic in the cells package that are to do with exporting. This is rather confusing. I think that the `FlowLogExporterBuilder` interface should rather be defined in `pkg/hubble/cell` and the logic implemented in `pkg/hubble/exporter` for the two different export implementations we have. That's also where we could more easily test the builder implementations. Does that make sense?